src/bun.js/api/bun/socket.zig

```diff
@@ -1352,33 +1350,19 @@ fn NewSocket(comptime ssl: bool) type {
 
             const args = callframe.arguments(4);
 
-            if (args.len == 0) {
-                log("end()", .{});
-                if (!this.detached) {
-                    if (!this.socket.isClosed()) this.socket.flush();
-                    this.detached = true;
-
-                    if (!this.socket.isClosed())
-                        this.socket.close(0, null);
```

Comment:
Won’t this prevent the socket from closing, leaking the network connection?

Reply:
It shouldn't - this logic just got folded into the one below, as call to `writeOrEnd()` with this condition would guarantee to trigger the clean-up code below.